source/async_stack.cpp

```diff
@@ -99,6 +99,11 @@ static std::uint64_t get_os_thread_id() {
   return tid;
 #  elif defined(_WIN32)
   return std::uint64_t(GetCurrentThreadId());
+#  elif defined(__FreeBSD__)
+  long tid;
+  thr_self(&tid);
+#  elif defined(__EMSCRIPTEN__)
+  return 0;
 #  else
   return std::uint64_t(gettid());
```

Comment:
Do you know whether you could express this `else` as an `elsif` that specifically targets the expected platform so the `else` branch could be an error?
```suggestion
#  elif defined(__linux__) // or whatever is actually correct
  return std::uint64_t(gettid());
#  else
#    error Unsupported platform in get_os_thread_id
```